Gpio driver

Surya-Santhosh opened the pull request:
Modified the TimerApp code for LED blink with delay of 840ms ON and 532ms OFF. Setup hardware and check the waveform in oscilloscope. Using RPi. Modified the same functions with a build macro to switch implementations between RPI and Native builds.

Surya-Santhosh committed:
Added Makefile

Surya-Santhosh committed:
Create CMakeLists.txt

Surya-Santhosh committed:
Changed minimum required version

Surya-Santhosh committed:
Add Cmakelists.txt  files for native and cross compiler.
Add common cmakelists.txt.
Add toolchain for Cross compiler.
Modify Readme file.

Surya-Santhosh committed:
Modify readme file

Surya-Santhosh committed:
Remove O2,O3  (Optimization) files

Surya-Santhosh committed:
Remove O1 (Optimization) file.

Surya-Santhosh committed:
Add GPIOControl folder to set the GPIO
Modify CMakeLists
Modify LEDDisplay - add new function to LED Blink
Modify main.c
Modify common.h

Surya-Santhosh committed:
Modified Makefile
modified LEDDisplay.h - add new line

adarshtrenser2023 commented on GPIOControl/GPIOControl.c:
```diff
@@ -0,0 +1,84 @@
+// Return  : blResult
+// Notes   : None
+//******************************************************************************
+bool GPIOControlSet(struct gpiod_chip **stpchip, 
```
double pointer needs pp as prefix , also check if st comes first or p comes first

adarshtrenser2023 commented on GPIOControl/GPIOControl.c:
```diff
@@ -0,0 +1,84 @@
+bool GPIOControlSet(struct gpiod_chip **stpchip, 
+                    struct gpiod_line **stpline)
+{
+    uint8 cState;
```
not initialized

adarshtrenser2023 commented on GPIOControl/GPIOControl.c:
```diff
@@ -0,0 +1,84 @@
+        blResult = false;
+    }
+
+    *stpline = gpiod_chip_get_line(*stpchip,GPIO_LINE);
```
add a space after ','

adarshtrenser2023 commented on main.c:
```diff
@@ -36,27 +40,49 @@ int main()
+    
     while (true)
     {
         time_t ulTime;
```
why declaring inside while declare it outside also initialize

adarshtrenser2023 commented on main.c:
```diff
@@ -36,27 +40,49 @@ int main()
     uint32 ulEpoch = 0; 
 
+    // Set GPIO.
+    #ifdef ENABLE_LED_BLINK
```
try to make macro meanig full like "ENABLE_LED_ON_RPI" or something that makes reader understand that this piece of code is only works for RPI

Surya-Santhosh committed:
Modify as per the review comments.
Change the cross compiler macro name to modify ENABLE_LED_ON_RPI.
 Add space after ','.
 Readme file - add LED Blink feature in description.

Surya-Santhosh committed:
Modified as per the rview comments.
Corrected prefix of variable..

Surya-Santhosh committed:
In .gitignore - add .txt
In CmakeLists.txt - add -g.

Surya-Santhosh committed:
Changed GPIO pin
